das-dashboard/backend/shared/internal/web_configuration.py

```
@@ -41,90 +45,163 @@ def load_user_profile(self):
+            return
+
         if not os.path.exists(CONFIG_PATH):
+            if required:
+                raise ConfigurationFileLoadError(f"Configuration file not found: {CONFIG_PATH}")
             self.config_dictionary = {}
             return
 
         try:
-            with open(CONFIG_PATH, "r") as f:
-                config = json.load(f)
+            with open(CONFIG_PATH, "r") as config_file:
```

Ruff (0.15.21)

[warning] Unnecessary mode argument

Remove mode argument

(UP015)